pyomo/scripting/plugins/extras.py

```diff
@@ -30,6 +32,10 @@ def get_packages():
         packages.append(('pyro','Pyro'))
     return packages
 
+@deprecated(
+        "Use of the pyomo install-extras is deprecated. There are known bugs, "
+        "and we do not recommend the use of this code. ",
+        version='TBD')
```

Comment:
Might want to add an explanation that the current recommended course of action is to install optional dependencies when they're needed.